### ISSUE TITLE
Scaling Mask-RCNN

### ISSUE THREAD

**Issue by robieta** [mlcommons/training_policies#139]
Currently there is an ambiguity in the rules about how to scale Mask-RCNN. (Thanks Paulius for bringing this up.) Detectron hard codes 2000 candidate images sampled from the minibatch, which in the case of the reference is 2. There has been a proposal that for scale up the sensible figure is to sample `1000 * minibatch size` rather that 2000 regardless of batch size.

Related, the question of example mixing also becomes problematic at large batch sizes. The way that the reference currently behaves (at least from my understanding consulting with our experts) is that it samples 2k candidate patches from both images, and then selects 512 patches from both images. (repeating if less than 512 patches are selected from one image). This is problematic at large batch sizes both because it means that an image could have very few patches in the top k and therefore have a very high number of repeats (hurting convergence) and because it imposes a large coordination burden when distributing the minibatch across multiple nodes. We would therefore suggest that the scaling rules for Mask-RCNN be interpreted to allow training to respect image boundaries (selecting the best 1k from each image in the first pass). Our principle reason for not suggesting that the bounds of every 2 images be respected is that that would unnecessarily complicate the networks with little or no benefit.

Concretely, our proposal is:
"Submitters may sample either 2000 or 1000 * batch_size patches, and may sample either from a pool of all examples or individually and uniformly from each image."

**Comment by petermattson** [comment]
SWG:
Submitters may sample either (2000) or (1000 * batch_size patches), and may sample either (from a pool of all examples) or (individually and uniformly from each image).
Both choices are hyper-parameters.
